### ISSUE TITLE
Add XDG configuration

### ISSUE THREAD

**Issue by bkuhlmann** [whitesmith/rubycritic#376]
## Overview

I would like to propose using the the [XDG](https://www.alchemists.io/projects/xdg) specification for managing this gem's configuration which is a UNIX standard. Example:

``` 
# Global (default)
$HOME/.config/rubycritic/configuration.yml

# Local (alternative per project configuration -- Takes precedence over global configuration)
$HOME/projects/my_example/.config/rubycritic/configuration.yml
```

This can be easily solved by using the [XDG](https://www.alchemists.io/projects/xdg) gem as a dependency to this project. Even better, you can use the [Runcom](https://www.alchemists.io/projects/runcom) gem which is a superset of XDG for allowing users to have either a global and/or local configuration since XDG doesn't support local configurations by default.

Doing this work would resolve the following related issues:

- [Add settings file.](https://github.com/whitesmith/rubycritic/issues/30)
- [Add excluded paths/files.](https://github.com/whitesmith/rubycritic/issues/11)
- [Reek configuration not being picked up.](https://github.com/whitesmith/rubycritic/issues/166)
- [Configuration integrity](https://github.com/whitesmith/rubycritic/issues/303) - This one may or may not be resolved with this work but thought I'd mention it regardless.

## Screenshots/Screencasts

Here's an example of ruby code used in the [Rubysmith](https://www.alchemists.io/projects/rubysmith) project to s how how easy it would be to implement this:

![image](https://user-images.githubusercontent.com/50245/101987538-db0cd180-3c51-11eb-900f-bfeaabce1847.png)

:bulb: Associated source code can be found [here](https://github.com/bkuhlmann/rubysmith/blob/master/lib/rubysmith/cli/configuration.rb).

## Desired Behavior

By using [Runcom](https://www.alchemists.io/projects/runcom), which is built on top of [XDG](https://www.alchemists.io/projects/xdg), you'd get the following enhancements for free:

- Checks to see if local configuration exists and loads it.
- If local configuration doesn't exist, then it checks to see if global configuration exists and loads it. 
- If global configuration doesn't exist, then default Rubycritic configuration is loaded instead.

By doing this work, we'd have a way to remove the `.rubycritic.yml` from the root of every project thus preventing an extra file showing up in the root people's projects.

**Comment by go2null** [comment]
I also like a clean project root, and I go to some lengths to get apps to use the XDG dirs. 

However, for software dev tooling, the __XDG__ (user level) dirs may not be the best choice as the configs should live with the repos.  What we need more are repo-level dirs to keep the repo root clean and focused on the "business" logic.  Thus, supporting a `<repo>/.config/rubycritic.yaml` may be more what's required?

just my 2cents :)

**Comment by bkuhlmann** [comment]
Yep, that's already supported via [Runcom](https://alchemists.io/projects/runcom). I enjoy project level configurations as well which Runcom handles well. I use Runcom for most of my projects, especially CLIs.
